provider/status_change.json

```diff
@@ -0,0 +1,169 @@
+              },
+              "provider_id": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/provider_id",
+                "description": "The UUID for the Provider, unique within MDS",
+                "$ref": "common.json#/definitions/uuid"
+              },
+              "device_id": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/device_id",
+                "description": "A unique device ID in UUID format",
+                "$ref": "common.json#/definitions/uuid"
+              },
+              "vehicle_id": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/vehicle_id",
+                "type": "string",
+                "description": "The Vehicle Identification Number visible on the vehicle itself",
+                "default": "",
+                "examples": [
+                  "ABC123"
+                ],
+                "pattern": "^(.*)$"
+              },
+              "vehicle_type": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/vehicle_type",
+                "type": "string",
+                "description": "The type of vehicle",
+                "enum": [
+                  "bicycle",
+                  "scooter"
+                ]
+              },
+              "propulsion_type": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/propulsion_type",
+                "type": "array",
+                "description": "The type of propulsion; allows multiple values",
+                "items": {
+                  "type": "string",
+                  "enum": [
+                    "human",
+                    "electric",
+                    "combustion"
+                  ]
+                },
+                "minItems": 1
+              },
+              "event_type": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/event_type",
+                "type": "string",
+                "description": "The type of the event",
+                "enum": [
+                  "available",
+                  "reserved",
+                  "unavailable",
+                  "removed"
+                ]
+              },
+              "event_type_reason": {
+                "$id": "#/properties/data/properties/status_changes/items/properties/event_type_reason",
+                "type": "string",
+                "description": "The reason for the event",
+                "enum": [
+                  "service_start",
+                  "user_drop_off",
+                  "rebalance_drop_off",
+                  "maintenance_drop_off",
+                  "user_pick_up",
+                  "maintenance",
+                  "low_battery",
+                  "service_end",
+                  "rebalance_pick_up",
+                  "maintenance_pick_up"
+                ]
+              },
```

Comment:
For `event_type` and `event_type_reason`, it would be great if we could get even more granular with this schema to indicate that some `reasons` apply to some `event_types`, and others do not. I'm not sure how to represent that or if it's possible... we have like a hierarchical enum (taxonomy?) here 😄

Reply:
I *think* that should be possible. Let me look more closely at how to use some of these more complex `anyOf`/`allOf` stuff.

Reply:
I think I might punt on this for now -- I don't quite see a way to make `event_type_reason` contextually a different type. Maybe we can think on this for a follow-up?